pkg/plugins/golang/v4/scaffolds/internal/templates/makefile.go

```diff
@@ -183,7 +183,7 @@ run: manifests generate fmt vet ## Run a controller from your host.
 # More info: https://docs.docker.com/develop/develop-images/build_enhancements/
 .PHONY: docker-build
 docker-build: ## Build docker image with the manager.
-	$(CONTAINER_TOOL) build -t ${IMG} .
+	$(CONTAINER_TOOL) build --load -t ${IMG} .
```

Comment:
It is the default docker build it is working fine.
you probably want replace the `--flag push` with `--load` in the docker buildX instead, see: https://github.com/kubernetes-sigs/kubebuilder/pull/4547/files#diff-e5d6d4f9d67174f1f3764674eb786688353c77cef8514e24d1323e55e5a4a7e8R192-R209

However, the WARNING is only faced in the buildx(above target) when it has no flags `--load` or `--push`

Reply:
docker buildx do not require --load at all.. if you have --push, you can skip --load.

Reply:
Hi @kaovilai 

See: https://github.com/kubernetes-sigs/kubebuilder/issues/4546#issuecomment-2644025655

I reproduce your scenario:

You are calling `docker buildx` without ANY of the flags.
It is required to have --load or push

And this target here is docker build ( should not use any of those )
You can either validate the behaviour with the samples under testdata to check that all generated by Kubebuilder works fine.

So, I am closing this one as deferred if you do not mind

Reply:
@camilamacedo86 per https://github.com/kubernetes-sigs/kubebuilder/issues/4546#issuecomment-2644142851 this can be triggered with just docker-build per my reproduce. I did not modify the default scaffold.

Reply:
This is one of the PRs that benefits the users of the **default scaffold without modification.**